components/Buttons/examples/ButtonsTypicalUse.m

```diff
@@ -135,14 +136,26 @@ - (void)viewDidLoad {
 - (void)didTap:(id)sender {
   NSLog(@"%@ was tapped.", NSStringFromClass([sender class]));
   if (sender == self.floatingButton) {
-    [self.floatingButton
-          collapse:YES
-        completion:^{
-          dispatch_after(dispatch_time(DISPATCH_TIME_NOW, (int64_t)(1 * NSEC_PER_SEC)),
-                         dispatch_get_main_queue(), ^{
-                           [self.floatingButton expand:YES completion:nil];
-                         });
-        }];
+    [UIView animateWithDuration:0.25 animations:^{
+      self.floatingButton.shape += 1;
+      if (self.floatingButton.shape > MDCFloatingButtonShapeExtendedTrailingIcon) {
+        self.floatingButton.shape = 0;
+        static dispatch_once_t onceToken;
+        dispatch_once(&onceToken, ^{
+          MDCBasicColorScheme *scheme = [[MDCBasicColorScheme alloc] initWithPrimaryColor:UIColor.purpleColor
+                                                                           secondaryColor:UIColor.orangeColor];
+          [ExampleFloatingButtonThemer applyToButton:self.floatingButton withColorScheme:scheme];
+        });
+
+      }
+      if (self.floatingButton.shape >= MDCFloatingButtonShapeExtendedLeadingIcon) {
+        [self.floatingButton setTitle:@"Extended" forState:UIControlStateNormal];
+      } else {
+        [self.floatingButton setTitle:nil forState:UIControlStateNormal];
```

Comment:
So when we change shape to not extended we have to nil the title? what happens when we forget to do that?

Reply:
Maybe we should just hide the title?

Reply:
I think we should leave it up to clients to add/remove the title along with the shape:
* UIButton ignores the alpha/hidden properties of the titleLabel when performing layout. The imageView is still offset as though the title were visible.
* The title is set on a per-state basis, and the button probably should not be removing all possible states (including the UIControlStateApplication range).
* Compact FABs may use the titleLabel instead of an image in some cases. Some phrases with low character count may be preferable to images.

### Hidden title that contains text (includes frames)
![screen shot 2017-11-17 at 9 51 45 am](https://user-images.githubusercontent.com/1753199/32953181-7334ee1e-cb7d-11e7-825c-50d8dcace6e0.png)

### FAB with title-only in compact mode
![screen shot 2017-11-17 at 10 00 22 am](https://user-images.githubusercontent.com/1753199/32953426-2687c4c8-cb7e-11e7-8a51-0cf91602ca50.png)